src/Tools/Microsoft.dotnet-openapi/test/dotnet-microsoft.openapi.Tests.csproj

```diff
@@ -18,6 +18,7 @@
 
   <ItemGroup>
     <Reference Include="Microsoft.Build" ExcludeAssets="runtime" />
+    <Reference Include="NuGet.Frameworks" />
```

Comment:
@dougbu turns out the newer SDK expects support for the .NET 6.0 TFM. These tests evaluate MSBuild in-proc and we have to somehow squirrel an updated version to the test.

Reply:
Not sure how to parse this @pranavkm. Without a PR description, I have no way of knowing how the pieces fit together, beyond this change and the one in eng/Dependencies.props being linked. So, what's happening here❔

Reply:
Sorry, let me rephrase. 

* The openAPI tests host an MSBuild workspace in process. https://github.com/dotnet/aspnetcore/blob/main/src/Tools/Microsoft.dotnet-openapi/src/Commands/BaseCommand.cs#L129-L137 which results in an MSBuild evaluation of the project using the SDK we have installed.
* The most recent SDK expects a version of NuGet.Framework that supports the 6.0 TFM.
* Prior to this change, the NuGet.Versioning package was brought in transitively by Microsoft.NET.Sdk.Test SDK that's added to all test projects:
```
"Microsoft.TestPlatform.ObjectModel/16.6.1": {
        "type": "package",
        "dependencies": {
          "NuGet.Frameworks": "5.0.0"
        },
```
* Using the 5.0 version of the package results in the MSBuild evaluation and the test to fail. 
* Adding the reference and updating the package reference to a newer version fixes the problem.

Reply:
Btw, here's the error prior to this change:

```
Threw error: The expression "[MSBuild]::GetTargetFrameworkIdentifier(net6.0)" cannot be evaluated. MSB0001: Internal MSBuild Error: A required NuGet assembly was not found. Expected Path: /datadisks/disk1/work/9FD308A7/w/A22A0907/e/.dotnet24379/sdk/6.0.100-preview.2.21118.6  /datadisks/disk1/work/9FD308A7/w/A22A0907/e/.dotnet24379/sdk/6.0.100-preview.2.21118.6/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.TargetFrameworkInference.targets
```

Reply:
I see. So, we're basically working around out-of-date dependencies in the latest Microsoft.TestPlatform.ObjectModel packages. Ick but :ok: